src/Plugin/Field/FieldType/EntityVersionItem.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\entity_version\Plugin\Field\FieldType;
+
+use Drupal\Core\Field\FieldItemBase;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\TypedData\DataDefinition;
+
+/**
+ * Plugin implementation of the 'entity_version' field type.
+ *
+ * @FieldType(
+ *   id = "entity_version",
+ *   label = @Translation("Entity version"),
+ *   module = "entity_version",
+ *   description = @Translation("Stores the version of the entity.")
+ * )
+ */
+class EntityVersionItem extends FieldItemBase implements EntityVersionItemInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function schema(FieldStorageDefinitionInterface $field_definition) {
+    return [
+      'columns' => [
+        'major' => [
+          'type' => 'int',
+        ],
+        'minor' => [
+          'type' => 'int',
+        ],
+        'patch' => [
+          'type' => 'int',
+        ],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isEmpty() {
+    // We consider the field empty if either of these properties left empty.
+    $major = $this->get('major')->getValue();
+    $minor = $this->get('minor')->getValue();
+    $patch = $this->get('patch')->getValue();
+
+    return $major === NULL || $major === '' || $minor === NULL || $minor === '' || $patch === NULL || $patch === '';
```

Comment:
Can't we just check the three values are empty() ?

Reply:
No, because integer 0 is considered empty. and for us it is not empty.